src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -731,16 +731,20 @@ protected function isNested($name, $relation)
     /**
      * Apply the callback's query changes if the given "value" is true.
      *
-     * @param  bool  $value
-     * @param  \Closure  $callback
-     * @return $this
+     * @param  bool  $condition
+     * @param  \Closure|string  $callback
+     * @param  string  $operator
+     * @param  string  $value
+     * @return \Illuminate\Database\Query\Builder
      */
-    public function when($value, $callback)
+    public function when($condition, $callback, $operator = null, $value = null)
```

Comment:
The second argument should probably renamed to `$column`.

It makes more sense to check `is_callable($column` than this `$this->where($callback, $operator, $value)` stuff.